components/validation/validation-custom-mixin.js

```diff
@@ -0,0 +1,51 @@
+
+export const ValidationCustomMixin = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			failureText: { type: String, attribute: 'failure-text' },
```

Comment:
Does anyone have any thoughts on whether we want `failure-text` as a property or if we want to be more flexible by instead allowing a `failureText` string to be returned from `validate`?

Reply:
That's a good question. I'm trying to think of a scenario where a single validator could result in different failure messages, and I'm not coming up with a good one. Very often there might be different error conditions (e.g. number must be greater than 4 AND less than 10), but in those cases we'd just craft an error message to reflect both conditions I think (e.g. "Number must be between 4 and 10"). And worst case for very complex scenarios they could just use multiple custom validators?

Reply:
I was thinking there could be cases where server-side validation would be performed and the server would return an error message or enum but I can't think of any exact cases off the top of my head. Having multiple custom validators is always an option but it could get kind of cumbersome and would result in duplicate validation calls if its to a single API call with multiple messages.